### ISSUE TITLE
Not getting data from local Cosmos Db

### ISSUE THREAD

**Issue by MickCG** [IEvangelist/azure-cosmos-dotnet-repository#34]
First I am probably biting off more than I can chew. I have a fair amount of experience but all pretty simple apps in the past. Having said that, I stumbled on the package and cloned the repository. I looked at the Azure Function example specifically, since that is what I'm needing right now. My example that I was having trouble with redid the Startup, modified the data models to have the class from Item. installed the nugget package. Thought I am golden. The function compiles, throws no errors, looks in the cmd window that it succeeded, but returns a null from the GetAsync, the OkObjectResult shows a null. Anyway, I then tried to run the AzureFunctionTier on the solution cloned after making it work for my local Cosmos db. Can you point me in a direction of what in the world I could be doing wrong? This all looks dirt simple.

**Comment by IEvangelist** [comment]
Hi @MickCG - I'm thrilled that you found this and have been attempting to use it. Sorry to hear that you're not having much luck, with that said, let's work backwards a bit and make sure you have everything setup correctly.

1. Do you have an Azure Account created?
2. Do you have an Azure Cosmos DB resource created?
3. Did you correctly configure the settings with the Primary Key?

**Comment by MickCG** [comment]
I just screwing around with the function example had success. Man trying a bunch of things all at once just causes confusion you think I would know that from my previous life as a Computer Tech / Network admin. Anyway, found my problem, in the AzureFuctionTier the problem is happening in the get based on await _repository.GetAsync(u => u.EmailAddress == emailAddress, cancellationSource.Token seems to not parse correctly I suspect. When I changed to LastName, works like a champ. I think I need to dig deep into repository GetAsync? 

Oh and yes to all of the above.

**Comment by IEvangelist** [comment]
So can we close this issue? Or are you going to update it to state that the expression doesn't correctly get compiled?

**Comment by MickCG** [comment]
Oh sorry, yes I think I am off and running, close this issue. I swear I have messed around for the easiest way to build the backend with Azure Functions for the project I am working on, which will be a mobile app with Xamarin.Forms. I explored using EF for the job of accessing Cosmosm db. But ran into big problems with versions. This is a lot more straight forward all the way around. Thanks for your help!
